qutip/utilities.py

```diff
@@ -331,3 +342,178 @@ def _version2int(version_string):
         "post")[0].split('.')
     return sum([int(d if len(d) > 0 else 0) * (100 ** (3 - n))
                 for n, d in enumerate(str_list[:3])])
+
+
+# -----------------------------------------------------------------------------
+# Fitting utilities
+#
+
+def iterated_fit(
+    fun: Callable[..., complex], num_params: int,
+    xdata: ArrayLike, ydata: ArrayLike,
+    target_rmse: float = 1e-5,
+    guess: ArrayLike | Callable[[int], ArrayLike] = None,
+    Nmin: int = 1, Nmax: int = 10,
+    lower: ArrayLike = None, upper: ArrayLike = None,
+    sigma: float | ArrayLike = None, maxfev: int = None
```

Comment:
just a minor thing, previously maxfev was hardcoded.  i think its better as here, to use default and allow user to define, but  the default is insufficient for some of the paper examples, so just a note that when updating those examples we have to up it

Reply:
These parameters should also be on separate lines. It took me a few seconds to find `maxfev`.